### PR TITLE
Fix createExistingStake and other refactors/fixes

### PR DESCRIPTION
**Pull request by Mvii7**

### CODE REVIEW

File: contracts/stake/StakeManager.sol [hunk 1]
```diff
@@ -198,7 +180,7 @@ contract StakeManager is IStakeManager, StakeBase {
 
         uint256 newEnd = block.timestamp + (uint256(settings.secondsInDay) * 5555);
 
-        contracts.vcAuction.addTotalSharesOfAndRebalance(stakeUpgrade.staker, newShares);
+        contracts.vcAuction.addTotalSharesOfAndRebalance(stakeUpgrade.staker, newShares - stakeUpgrade.shares);
```
Comment (Mvii7) [inline contracts/stake/StakeManager.sol:183]: Need this right?

Comment (petec86) [reply contracts/stake/StakeManager.sol:183]: Yes

File: contracts/stake/StakeMinter.sol [hunk 2]
```diff
@@ -169,17 +169,16 @@ contract StakeMinter is IStakeMinter, Pausable, ExternallyCallable {
             require(shares != 0, 'STAKE BURNER: stake withdrawn on v1.');
         }
 
-        StakeData memory newStake =
-            contracts.stakeManager.createExistingStake(amount, shares, start, end, id);
+        contracts.stakeManager.createExistingStake(id, amount, shares, start, end, firstInterestDay);
 
         /** Since we are updating from v1 or v2 we need to rebalance the shares. */
         if (from != to) {
```
Comment (Mvii7) [inline contracts/stake/StakeMinter.sol:175]: Wtf is going on here? Both blocks are same except for withdraw

Comment (Mvii7) [reply contracts/stake/StakeMinter.sol:175]: Else is not needed right?

Comment (Mvii7) [reply contracts/stake/StakeMinter.sol:175]: if (from != to)

File: contracts/stake/StakeMinter.sol [hunk 3]
```diff
@@ -169,17 +169,12 @@ contract StakeMinter is IStakeMinter, Pausable, ExternallyCallable {
...
             contracts.vcAuction.withdrawDivTokensFromTo(from, to);
```
Comment (Mvii7) [inline contracts/stake/StakeMinter.sol:176]: Do we need to withdraw div tokens from - to? @petec86 isn't rebalance enough?

File: contracts/stake/StakeMinter.sol (hunk 2 above)
Comment (petec86) [reply contracts/stake/StakeMinter.sol:175]: is there any reason we actually have this option, for from to be != to?

File: contracts/stake/StakeMinter.sol (hunk 3 above)
Comment (petec86) [reply contracts/stake/StakeMinter.sol:176]: Rebalance is enough. Plus, if you think about, it makes no sense to transfer ALL earned divs to the "to" address. Imagine I want to mint a stake as NFT in order to send it to someone else; he would also receive all my earned divs.

File: contracts/stake/StakeMinter.sol (hunk 2 above)
Comment (petec86) [reply contracts/stake/StakeMinter.sol:175]: correct. we only need to rebalance if the stake moves to another account "to".
else, we don't need to do anything.

**Review by petec86 — APPROVED** [review]

File: contracts/stake/StakeMinter.sol (hunk 3 above)
Comment (Mvii7) [reply contracts/stake/StakeMinter.sol:176]: We need to implement withdraw by %, and withdraw that % on NFT send. Or just withdraw everything?

Comment (petec86) [reply contracts/stake/StakeMinter.sol:176]: There's no need for an withdraw at all. If you want an withdraw, simply withdraw everything. That percentage thing would not be accurate.

File: contracts/stake/StakeToken.sol [hunk 4]
```diff
@@ -84,8 +84,8 @@ contract StakeToken is
             'TOKEN: Stake is withdrawn and can not be transferred'
         );
 
-        contracts.vcAuction.subTotalSharesOfAndRebalance(from, stake.shares);
-        contracts.vcAuction.addTotalSharesOfAndRebalance(to, stake.shares);
+        contracts.vcAuction.withdrawDivTokensFromTo(from, from);
```
Comment (Mvii7) [inline contracts/stake/StakeToken.sol:87]: Need to look at this, ideally the 2 funcs should be 1 func

Comment (Mvii7) [reply contracts/stake/StakeToken.sol:87]: The caveat here is that everything is withdrawn on NFT transfer, which is fine

File: contracts/stake/StakeMinter.sol (hunk 3 above)
Comment (Mvii7) [reply contracts/stake/StakeMinter.sol]: There is a need to withdraw as you said, you don't want to transfer accrued divs

Comment (Mvii7) [reply contracts/stake/StakeMinter.sol]: OK so just rebalance is enough as divs will be correct in both accounts and wont move

File: contracts/stake/StakeToken.sol (hunk 4 above)
Comment (Mvii7) [reply contracts/stake/StakeToken.sol]: Dont need this, just rebalance is enough

File: contracts/auction/VCAuction.sol [hunk 5]
```diff
@@ -91,16 +91,6 @@ contract VCAuction is IVCAuction, Manageable, Migrateable, ExternallyCallable {
         withdrawDivTokenInternal(msg.sender, to, tokenAddress);
     }
 
-    function withdrawDivTokensFromTo(address from, address payable to)
```
Comment (petec86) [inline contracts/auction/VCAuction.sol:94]: might want to keep this. The idea was to offer the option (feature) to withdraw divs to other address than the main account address.

Comment (Mvii7) [reply contracts/auction/VCAuction.sol]: yup

File: contracts/stake/StakeMinter.sol [hunk 6]
```diff
@@ -169,17 +169,12 @@ contract StakeMinter is IStakeMinter, Pausable, ExternallyCallable {
...
         }
 
-        StakeData memory newStake =
-            contracts.stakeManager.createExistingStake(amount, shares, start, end, id);
+        contracts.stakeManager.createExistingStake(id, amount, shares, start, end, firstInterestDay);
 
-        /** Since we are updating from v1 or v2 we need to rebalance the shares. */
+        /** Since we are updating from v1 or v2 we need to rebalance the shares if not same wallet. */
         if (from != to) {
             contracts.vcAuction.withdrawDivTokensFromTo(from, to);
```
Comment (Mvii7) [inline contracts/stake/StakeMinter.sol:176]: Can make these 2 1 func, but only manager will be getting into this block

**Pull request merged**